Sources/ArcGISToolkit/Components/Augmented Reality/Utilities/WorldTrackingSceneView.swift

```diff
@@ -16,74 +16,94 @@ import ARKit
+    /// A Boolean value that indicates if the user is calibrating.
+    private let isCalibrating: Bool
+    /// The location datasource that is used to access the device location.
+    private let locationDataSource: LocationDataSource
+    /// The closure that builds the scene view.
+    private let sceneViewBuilder: (SceneViewProxy) -> SceneView
+    /// The time threshold in seconds between location updates to reset the world-tracking session.
+    private let timeThreshold: Double
     /// The proxy for the ARSwiftUIView.
     @State private var arViewProxy = ARSwiftUIViewProxy()
     /// The camera controller that will be set on the scene view.
     @State private var cameraController: TransformationMatrixCameraController
-    /// A Boolean value that indicates if the user is calibrating.
-    @State var isCalibrating = false
-    /// The current interface orientation.
-    @State private var interfaceOrientation: InterfaceOrientation?
-    /// The location datasource that is used to access the device location.
-    @State private var locationDataSource: LocationDataSource
-    /// A Boolean value indicating if the camera was initially set.
-    @State private var initialCameraIsSet = false
     /// A Boolean value that indicates whether the coaching overlay view is active.
     @State private var coachingOverlayIsActive = false
     /// The current camera of the scene view.
     @State private var currentCamera: Camera?
-    /// A Boolean value that indicates whether the calibration view is hidden.
-    private var calibrationViewIsHidden = false
-    /// The closure that builds the scene view.
-    private let sceneViewBuilder: (SceneViewProxy) -> SceneView
-    /// The configuration for the AR session.
-    private let configuration: ARConfiguration
-    /// The timestamp of the last received location.
-    @State private var lastLocationTimestamp: Date?
     /// The current device location.
     @State private var currentLocation: Location?
-    /// The valid accuracy threshold for a location in meters.
-    private var validAccuracyThreshold = 0.0
-    /// Determines the alignment of the calibration view.
-    private var calibrationViewAlignment: Alignment = .bottom
-    /// The view model for the calibration view.
-    @StateObject private var calibrationViewModel = CalibrationViewModel()
+    /// The current interface orientation.
+    @State private var interfaceOrientation: InterfaceOrientation?
+    /// The timestamp of the last received location.
+    @State private var lastLocationTimestamp: Date?
+    
+    /// Projected point from the spatial reference of the location data source's to the scene view's.
+    private var currentPosition: Point? {
+        guard let currentLocation,
+              let currentCamera,
+              let spatialReference = currentCamera.location.spatialReference,
+              let position = GeometryEngine.project(currentLocation.position, into: spatialReference)
```

Comment:
I'm not sure if we should also use the elevation (z value) from the projected point, but it doesn't make a big difference in my tests.

Reply:
Maybe something to ask Koushik.

Reply:
Actually I'm seeing a difference between the 3D point's z value from the System LDS, and the point after I project it into the scene's SR. I'll reach out and ask.

Reply:
After talking to Koushik, I'm pretty sure we can use the z value from GPS directly as both Apple and us are representing it in MSL. I'll remove this logic